### ISSUE TITLE
remote_user on task doesn't work

### ISSUE THREAD

**Issue by bastitx** [ansible/ansible#13113]
Issue Type: Bug Report

Ansible Version:

```
ansible 2.0.0 (devel a99f2fd52f) last updated 2015/11/10 15:46:00 (GMT +000)
  lib/ansible/modules/core: (detached HEAD fde6ee15d6) last updated 2015/11/10 15:51:18 (GMT +000)
  lib/ansible/modules/extras: (detached HEAD 6cc43bbd56) last updated 2015/11/10 15:51:18 (GMT +000)

```

Ansible Configuration: Standard

Environment: AIX (Control machine & remote host)

Summary: When setting a different remote user on a task, it doesn't actually switch to it.

Steps To Reproduce:

```

---
- name: "test play"
  hosts: all
  remote_user: user1
  tasks:
    - shell: whoami
      register: result
      remote_user: user2
    - debug: msg={{result.stdout}}
```

Expected Results: Output should be "user2"

Actual Results: Output is "user1"

**Comment by amenonsen** [comment]
I can confirm that this is not specific to AIX. Setting `ansible_user` in the inventory or setting `remote_user` on a play both work for me, but setting `remote_user` on a task or block just doesn't work (and I checked; `playbooks_intro.rst` says clearly that it should work).

**Comment by trkoch** [comment]
I also observed this issue. As @amenonsen mentioned the docs are pretty clear about this.

> Remote users can also be defined per task.
> http://docs.ansible.com/ansible/playbooks_intro.html#hosts-and-users

```
- name: Debug
  remote_user: root
  shell: whoami
```

`whoami` reports the globally defined remote user, not root.

**Comment by zentavr** [comment]
When I have a code like (`remote_user` is in task):

```
- name: Renaming nodes via OS calls
  hosts: tag_deployment_ansible
  gather_facts: False
  accelerate: "{{ do_accelerate|default(False) }}"
  tasks:
    - name: Setting up hostname via the OS calls
      become: True
      remote_user: "{{ idle_user }}"
      hostname: name={{ hostnamevar|default(ec2_tag_Name) }}
      tags: [ 'rename' ]
      when: ec2_tag_environment == game_env and ec2_tag_provisioned is not defined

    - name: Renaming the machine (setting up EC2 tag)
      local_action:
        module: ec2_tag
        resource: "{{ ec2_id }}"
        region: "{{ ec2_region }}"
        state: present
        args:
          tags:
            Name: "{{ hostnamevar }}"
            provisioned: True
            glu_agent_name: "{{ hostnamevar }}-{{ ec2_id }}"
      tags: [ 'rename' ]
      when: ec2_tag_environment == game_env and ec2_tag_provisioned is not defined
```

The SSH connection tries to happen using my current username in local system, it does not take it from the variable. So I was forced to rebuild the code in another way and move that settings upper:

```
- name: Renaming nodes via OS calls
  hosts: tag_deployment_ansible
  gather_facts: False
  accelerate: "{{ do_accelerate|default(False) }}"
  remote_user: "{{ idle_user }}"
  tasks:
    - name: Setting up hostname via the OS calls
      become: True
      hostname: name={{ hostnamevar|default(ec2_tag_Name) }}
      tags: [ 'rename' ]
      when: ec2_tag_environment == game_env and ec2_tag_provisioned is not defined

    - name: Renaming the machine (setting up EC2 tag)
      local_action:
        module: ec2_tag
        resource: "{{ ec2_id }}"
        region: "{{ ec2_region }}"
        state: present
        args:
          tags:
            Name: "{{ hostnamevar }}"
            provisioned: True
            glu_agent_name: "{{ hostnamevar }}-{{ ec2_id }}"
      tags: [ 'rename' ]
      when: ec2_tag_environment == game_env and ec2_tag_provisioned is not defined
```

**Comment by abadger** [comment]
It was introduced by this performance fix: 63c47fb2715311a1a6258332221034585ea23593

Previous commit was good: c64ac90560938811fac4ff32ae69e1e667dddc34

**Comment by jimi-c** [comment]
# Closing This Ticket

Hi!

We believe the above commit should resolve this problem for you. This will also be included in the next major release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!
